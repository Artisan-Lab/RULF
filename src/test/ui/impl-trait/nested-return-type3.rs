// check-pass

trait Duh {}

impl Duh for i32 {}

trait Trait {
    type Assoc: Duh;
}

impl<F: Duh> Trait for F {
    type Assoc = F;
}

fn foo() -> impl Trait<Assoc = impl Send> {
    //~^ WARN opaque type `impl Trait<Assoc = impl Send>` does not satisfy its associated type bounds
    42
}

fn main() {
}
