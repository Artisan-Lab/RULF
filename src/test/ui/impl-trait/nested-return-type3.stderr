warning: opaque type `impl Trait<Assoc = impl Send>` does not satisfy its associated type bounds
  --> $DIR/nested-return-type3.rs:15:24
   |
LL |     type Assoc: Duh;
   |                 --- this associated type bound is unsatisfied for `impl Send`
...
LL | fn foo() -> impl Trait<Assoc = impl Send> {
   |                        ^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(opaque_hidden_inferred_bound)]` on by default
help: add this bound
   |
LL | fn foo() -> impl Trait<Assoc = impl Send + Duh> {
   |                                          +++++

warning: 1 warning emitted

